src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -2347,6 +2356,20 @@ void EmitNegativeLookaroundAssertion(RegexNode node)
             // Emits the code for the node.
             void EmitNode(RegexNode node, RegexNode? subsequent = null, bool emitLengthChecksIfRequired = true)
             {
+                // Before we handle general-purpose matching logic for nodes, handle any special-casing.
+                // -
+                if (_regexTree!.FindOptimizations.FindMode == FindNextStartingPositionMode.LiteralAfterLoop_LeftToRight_CaseSensitive &&
+                    _regexTree!.FindOptimizations.LiteralAfterLoop?.LoopNode == node)
```

Comment:
Adding a comment why supressing the _regexTree nullable warning would be imo useful.
How do we know that _regexTree is not null?